samples/Exporters/Console/TestRedis.cs

```diff
@@ -31,32 +31,32 @@ internal static object Run(string zipkinUri)
             var connection = ConnectionMultiplexer.Connect("localhost:6379");
 
             // Configure exporter to export traces to Zipkin
-            using var tracerFactory = TracerFactory.Create(builder => builder
-                .UseZipkin(o =>
-                {
-                    o.ServiceName = "redis-test";
-                    o.Endpoint = new Uri(zipkinUri);
-                })
-                .AddInstrumentation(t =>
-                {
-                    var instrumentation = new StackExchangeRedisCallsInstrumentation(t);
-                    connection.RegisterProfiler(instrumentation.GetProfilerSessionsFactory());
-                    return instrumentation;
-                }));
-            var tracer = tracerFactory.GetTracer("redis-test");
-
-            // select a database (by default, DB = 0)
-            var db = connection.GetDatabase();
-
-            // Create a scoped span. It will end automatically when using statement ends
-            using (tracer.StartActiveSpan("Main", out _))
-            {
-                Console.WriteLine("About to do a busy work");
-                for (var i = 0; i < 10; i++)
-                {
-                    DoWork(db, tracer);
-                }
-            }
+            // using var tracerFactory = TracerFactory.Create(builder => builder
+            //     .UseZipkin(o =>
+            //     {
+            //         o.ServiceName = "redis-test";
+            //         o.Endpoint = new Uri(zipkinUri);
+            //     })
+            //     .AddInstrumentation(t =>
+            //     {
+            //         var instrumentation = new StackExchangeRedisCallsInstrumentation(t);
+            //         connection.RegisterProfiler(instrumentation.GetProfilerSessionsFactory());
+            //         return instrumentation;
+            //     }));
+            // var tracer = tracerFactory.GetTracer("redis-test");
+            // 
+            // // select a database (by default, DB = 0)
+            // var db = connection.GetDatabase();
+            // 
+            // // Create a scoped span. It will end automatically when using statement ends
+            // using (tracer.StartActiveSpan("Main", out _))
+            // {
+            //     Console.WriteLine("About to do a busy work");
+            //     for (var i = 0; i < 10; i++)
+            //     {
+            //         DoWork(db, tracer);
+            //     }
+            // }
 
```

Comment:
I don't know how to port this to Activity style

Reply:
I took a stab at it. Below is where I landed.

Problem is this: `var instrumentation = new StackExchangeRedisCallsInstrumentation(t);`

Redis instrumentation hasn't been updated for ActivitySource yet, it still needs a Tracer. We should fix that up and then this should work as a replacement?

```csharp
// <copyright file="TestRedis.cs" company="OpenTelemetry Authors">
// Copyright The OpenTelemetry Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// </copyright>
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Threading;
using OpenTelemetry.Instrumentation.StackExchangeRedis;
using OpenTelemetry.Trace;
using OpenTelemetry.Trace.Configuration;
using StackExchange.Redis;

namespace Samples
{
    internal class TestRedis
    {
        internal static object Run(string zipkinUri)
        {
            // connect to the server
            var connection = ConnectionMultiplexer.Connect("localhost:6379");

            // Configure exporter to export traces to Zipkin
            using var openTelemetry = OpenTelemetrySdk.EnableOpenTelemetry(
                builder => builder
                    .UseZipkinActivityExporter(o =>
                    {
                        o.ServiceName = "redis-test";
                        o.Endpoint = new Uri(zipkinUri);
                    })
                    .AddInstrumentation(t =>
                    {
                        var instrumentation = new StackExchangeRedisCallsInstrumentation(t);
                        connection.RegisterProfiler(instrumentation.GetProfilerSessionsFactory());
                        return instrumentation;
                    })
                    .AddActivitySource("redis-test"));

            ActivitySource activitySource = new ActivitySource("redis-test");

            // select a database (by default, DB = 0)
            var db = connection.GetDatabase();

            // Create a scoped activity. It will end automatically when using statement ends
            using (activitySource.StartActivity("Main"))
            {
                Console.WriteLine("About to do a busy work");
                for (var i = 0; i < 10; i++)
                {
                    DoWork(db, activitySource);
                }
            }

            return null;
        }

        private static void DoWork(IDatabase db, ActivitySource activitySource)
        {
            // Start another activity. If another activity was already started, it'll use that activity as the parent activity.
            // In this example, the main method already started a activity, so that'll be the parent activity, and this will be
            // a child activity.
            using (Activity activity = activitySource.StartActivity("DoWork"))
            {
                try
                {
                    db.StringSet("key", "value " + DateTime.Now.ToLongDateString());

                    Console.WriteLine("Doing busy work");
                    Thread.Sleep(1000);

                    // run a command, in this case a GET
                    var myVal = db.StringGet("key");

                    Console.WriteLine(myVal);
                }
                catch (ArgumentOutOfRangeException e)
                {
                    // Set status upon error
                    activity.AddTag("ot.status", SpanHelper.GetCachedCanonicalCodeString(Status.Internal.CanonicalCode));
                    activity.AddTag("ot.status_description", e.ToString());
                }

                // Annotate our activity to capture metadata about our operation
                var attributes = new Dictionary<string, object>
                {
                    { "use", "demo" },
                };
                activity.AddEvent(new ActivityEvent("Invoking DoWork", attributes));
            }
        }
    }
}
```

Reply:
ok, i will update based on your example and leave the instrumentation commented with a TODO

Reply:
thanks for the example! still learning the idea of Span vs. Activity

Reply:
No problem! The code before was creating its own `Tracer` and then using that to emit spans. That's how we would before manually instrument some section of code. In the new `ActivitySource` world you have to add a custom source, tell OT to listen to it, and then you can spin up whatever manual activities you want using the source. Basically the key to it is this part:
```csharp
            using var openTelemetry = OpenTelemetrySdk.EnableOpenTelemetry(
                builder => builder
                    .AddActivitySource("redis-test"));

            ActivitySource activitySource = new ActivitySource("redis-test");
```

Reply:
thank you again for the explanation!